### PR TITLE
Fix a couple BioSequence bugs

### PR DESCRIPTION
**Pull request by dcjones**
I ran into a couple bugs with sequences. Test cases to reproduce and fixes here.

### CODE REVIEW

**Comment by dcjones** [comment]
Ohh don't merge this yet. The second commit breaks a bunch of other tests.

**Comment by dcjones** [comment]
Alright, got it figured out. These are fairly critical fixes so I'll merge later today if no one stops me.

**Comment by Ward9250** [comment]
What was the problem?

**Comment by dcjones** [comment]
Basically it's extremely easy for sub-sequences to get corrupted due to bugs in basic functions.
```julia

julia> seq = dna"ACGTACGTACGT";

julia> subseq = seq[3:6]
4nt DNA Sequence:
GTAC

julia> copy(subseq) # should be the same as subseq
4nt DNA Sequence:
ACGT
```

```julia

julia> seq = repeat(dna"ACGT", 8);

julia> subseq = seq[16:17]
2nt DNA Sequence:
TA

julia> Seq.orphan!(subseq) # should not change the sequence
2nt DNA Sequence:
T-
```

File: src/seq/bioseq.jl [hunk 1]
```diff
@@ -554,7 +554,7 @@ end
 function Base.copy{A}(seq::BioSequence{A})
     # NOTE: no need to set `seq.shared = true` here
     # since `newseq` will be `orphan!`ed soon.
-    newseq = BioSequence{A}(seq.data, 1:endof(seq), true)
+    newseq = BioSequence{A}(seq, 1:endof(seq))
```
Comment (bicycle1885) [inline src/seq/bioseq.jl]: Can you delete the two lines of comments above?

Comment (dcjones) [inline src/seq/bioseq.jl]: Yeah, done.

**Comment by bicycle1885** [comment]
Thank you!

**Pull request merged**